src/AzureCacheOptionsProviderWithToken.cs

```diff
@@ -36,7 +37,7 @@ internal class AzureCacheOptionsProviderWithToken : AzureCacheOptionsProvider, I
 
         if (string.IsNullOrWhiteSpace(azureCacheOptions.PrincipalId))
         {
-            throw new ArgumentException("Principal ID of a managed identity or service principal must be specified", nameof(azureCacheOptions.PrincipalId));
+            throw new ArgumentException("Principal ID of a managed identity, token credential, or service principal must be specified", nameof(azureCacheOptions.PrincipalId));
```

Comment:
The TokenCredential actually maps to an MI or SPN - I don't think we can talk about it having a Principal ID